wg-checkpoint-restore/charter.md

```diff
@@ -0,0 +1,90 @@
+
+# WG Checkpoint Restore Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+## Scope
+
+The Checkpoint/Restore Working Group aims to solve the problem of transparently
+checkpointing and restoring workloads in Kubernetes, a functionality discussed
+for over five years. The group will deliver the design and implementation of
+Checkpoint/Restore functionality in Kubernetes, serving as a central hub for
+community information and discussion. This initiative addresses a wide range of
+problems, including fault tolerance, improved resource utilization, and
+accelerated application startup times.
+
+### In scope
+
+- Identify core Kubernetes checkpoint/restore use cases (e.g., live migration,
+  fault tolerance, debugging, snapshotting) and gather stakeholder requirements.
+- Investigate and propose Kubernetes APIs for checkpoint/restore operations.
+- Work with SIGs for the best integration of checkpoint/restore functionality
+  and APIs.
+- Provide guidance for developers on checkpoint-friendly app design and
```

Comment:
there may be API needed to communicate between the app and API server that the checkopoint is requested AND/OR  that the app is ready for checkpoint